web3/src/main/java/org/hiero/mirror/web3/state/keyvalue/AccountReadableKVState.java

```diff
@@ -61,17 +63,32 @@ public AccountReadableKVState(
     @Override
     protected Account readFromDataSource(@Nonnull AccountID key) {
         final var timestamp = ContractCallContext.get().getTimestamp();
-        return commonEntityAccessor
+        final var account = commonEntityAccessor
                 .get(key, timestamp)
                 .filter(entity -> entity.getType() != TOKEN)
                 .map(entity -> {
-                    final var account = accountFromEntity(entity, timestamp);
+                    final var acc = accountFromEntity(entity, timestamp);
                     // Associate the account alias with this entity in the cache, if any.
-                    if (account.alias().length() > 0) {
-                        aliasedAccountCacheManager.putAccountAlias(account.alias(), key);
+                    if (acc.alias().length() > 0) {
+                        aliasedAccountCacheManager.putAccountAlias(acc.alias(), key);
                     }
-                    return account;
+                    return acc;
                 })
                 .orElse(null);
+        // In case a system account doesn't exist in a historical contract call
+        // return a dummy account to avoid errors like
+        // "Non-zero net hbar change when handling body"
+        if (account == null && isSystemAccount(key)) {
+            return Account.newBuilder().accountId(key).tinybarBalance(0).build();
+        }
+        return account;
+    }
+
+    private boolean isSystemAccount(AccountID key) {
+        if (key.hasAccountNum()) {
+            return key.accountNum() >= SYS_ACCOUNT_NUMBER_LOWER_LIMIT
+                    && key.accountNum() <= SYS_ACCOUNT_NUMBER_UPPER_LIMIT;
```

Comment:
This logic is off by one compared to `systemAccountDetector` bean. We should just check upper bound and use same constant for both.

```suggestion
            return key.accountNum() <= ServicesConfiguration.STRICT_SYSTEM_ACCOUNT_BOUNDARY;
```

Reply:
Addressed, the systemAccountDetector is now used.